service_linux.go

```diff
@@ -381,12 +390,24 @@ func programIngress(gwIP net.IP, ingressPorts []*PortConfig, isDelete bool) erro
 		}
 	}
 
+	handleIngressPorts = true
 	for _, iPort := range ingressPorts {
+
+		var portErr error
+
+		defer func() {
+			if portErr != nil && !isDelete {
+				rollbackPort := make([]*PortConfig, 0, 1)
+				rollbackPort = append(rollbackPort, iPort)
+				filterPortConfigs(rollbackPort, !isDelete)
+			}
+		}()
+
```

Comment:
I don't think that this block was ever needed because you were already running `filterPortConfigs(ingressPorts, !isDelete)` above.  But that's kind of moot if we move the `filterPortConfigs()` "undo" out to the caller.